packages/grpc-js/src/call-stream.ts

```diff
@@ -67,59 +64,143 @@ export interface WriteObject {
   flags?: number;
 }
 
+export interface MetadataListener {
+  (metadata: Metadata, next: (metadata: Metadata) => void): void;
+}
+
+export interface MessageListener {
+  // tslint:disable-next-line no-any
+  (message: any, next: (message: any) => void): void;
+}
+
+export interface StatusListener {
+  (status: StatusObject, next: (status: StatusObject) => void): void;
+}
+
+export interface FullListener {
+  onReceiveMetadata: MetadataListener;
+  onReceiveMessage: MessageListener;
+  onReceiveStatus: StatusListener;
+}
+
+export type Listener = Partial<FullListener>;
+
 /**
- * This interface represents a duplex stream associated with a single gRPC call.
+ * An object with methods for handling the responses to a call.
  */
-export type Call = {
+export interface InterceptingListener {
```

Comment:
That name comes from the original implementation in the other library, and the behavior of that class is basically the same. I agree that it's not the clearest name, but I would rather keep it so that it's easier to see the parallelism between the two implementations.